awsbestpractices/missing-pagination/MissingPagination.java

```diff
@@ -0,0 +1,507 @@
+    public void autoPaginatedMapperNotMissing_1() {
+        System.out.println(mapper.scan(DynamoSessionItem.class, new DynamoDBScanExpression()));
+    }
+
+    // Not missing - only need the firsts
+    public long onlyFirst_notMissing_1(final String materialName) {
+        final List<Map<String, AttributeValue>> items = dynamoClient.query(
+                new QueryRequest()
+                        .withTableName(tableName)
+                        .withConsistentRead(Boolean.TRUE)
+                        .withKeyConditions(
+                                Collections.singletonMap(
+                                        DEFAULT_HASH_KEY,
+                                        new Condition().withComparisonOperator(
+                                                ComparisonOperator.EQ).withAttributeValueList(
+                                                new AttributeValue().withS(materialName))))
+                        .withLimit(1).withScanIndexForward(false)
+                        .withAttributesToGet(DEFAULT_RANGE_KEY)).getItems();
+        if (items.isEmpty()) {
+            return -1L;
+        } else {
+            return Long.parseLong(items.get(0).get(DEFAULT_RANGE_KEY).getN());
+        }
+    }
+
+    // Not missing - stack name is set!
+    public String oneElementNotMissing_1() throws Exception {
+
+        DescribeStacksRequest wait = new DescribeStacksRequest();
+        wait.setStackName(deployment.getCloudFormationStackName());
+        System.out.println(new AmazonCloudFormation().describeStacks(wait).getStacks());
+    }
+
+    // Not missing - stack name is set (#2)
+    public void oneElementNotMissing_2() {
+        AmazonEC2 client = null;
+
+        try {
+            client = getEc2Client();
+            DescribeSecurityGroupsResult result;
+            DescribeSecurityGroupsRequest describeSecurityGroupsRequest =
+                    new DescribeSecurityGroupsRequest().withGroupIds(config.getACLGroupIdForVPC());
+            result = client.describeSecurityGroups(describeSecurityGroupsRequest);
+        } finally {
+            if (client != null) {
+                client.shutdown();
+            }
+        }
+    }
+
+    // Missing - describe without restrictions
+    public void describeNoFilters_1() {
+        AmazonEC2 client = null;
+
+        try {
+            client = getEc2Client();
+            DescribeSecurityGroupsResult result;
+            DescribeSecurityGroupsRequest describeSecurityGroupsRequest = new DescribeSecurityGroupsRequest();
+            result = client.describeSecurityGroups(describeSecurityGroupsRequest);
+            System.out.println(result.getSecurityGroups());
+        } finally {
+            if (client != null) {
+                client.shutdown();
+            }
+        }
+    }
+
+    // Missing
+    public void listAndStream_1() {
+        DescribeStacksResponse stacks =
+                cloudformation.describeStacks(DescribeStacksRequest.builder().build());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.